tests/unit/preprocessor/_area/test_area.py

```diff
@@ -15,6 +15,83 @@
                                            extract_region, extract_shape)
 
 
+def calc_grid_area():
+    """
+    Creates a regular grid with the same size as the fx file below.
+    """
+    coord_sys = iris.coord_systems.GeogCS(
+        iris.fileformats.pp.EARTH_RADIUS)
+    data = np.ones((2, 2))
+    lons = iris.coords.DimCoord(
+        [-5., 5.],
+        standard_name='longitude',
+        bounds=[[-10., 0.], [0., 10.]],
+        units='degrees_east',
+        coord_system=coord_sys)
+    lats = iris.coords.DimCoord(
+        [-5., 5.],
+        standard_name='latitude',
+        bounds=[[-10., 0.], [0., 10.]],
+        units='degrees_north',
+        coord_system=coord_sys)
+    coords_spec = [(lats, 0), (lons, 1)]
+    grid = iris.cube.Cube(data, dim_coords_and_dims=coords_spec)
+    return iris.analysis.cartography.area_weights(grid)
+
+
+def make_fx_file():
+    """
+    """
+    i_axis = iris.coords.DimCoord(
+        [i for i in range(2)],
+        standard_name=None,
+        var_name='i',
+        long_name='cell index along first dimension'
+        )
+    j_axis = iris.coords.DimCoord(
+        [i for i in range(2)],
+        standard_name=None,
+        var_name='j',
+        long_name='cell index along second dimension'
+        )
+
+    lat2d = np.array([[-5., -5.], [5., 5.]])
+    lon2d = np.array([[-5.,  5.], [-5., 5.]])
+
+    lat2d_bounds = np.array([[[-10., -10., 0., 0.],
+                              [-10., -10., 0., 0.]],
+                             [[0., 0., 10., 10.],
+                              [0., 0., 10., 10.]]])
+    lon2d_bounds = np.array([[[-10., 0., 0., -10.],
+                              [0., 10., 10., 0.]],
+                             [[-10., 0., 0., -10.],
+                             [0., 10., 10., 0.]]])
+
+    coords_spec = [(i_axis, 0), (j_axis, 1)]
+    data = calc_grid_area()
+
+    gridir = iris.cube.Cube(data, var_name='areacello', units=Unit('m^2'),
+                            dim_coords_and_dims=coords_spec)
+    print(gridir.data.mean())
+
+    lat_coord = iris.coords.AuxCoord(lat2d, standard_name='latitude',
+                                     long_name='latitude',
+                                     var_name='latitude',
+                                     units=Unit('degrees'),
+                                     bounds=lat2d_bounds)
+    lon_coord = iris.coords.AuxCoord(lon2d, standard_name='longitude',
+                                     long_name='longitude',
+                                     var_name='longitude',
+                                     units=Unit('degrees'),
+                                     bounds=lon2d_bounds)
+
+    gridir.add_aux_coord(lat_coord, data_dims=(0, 1))
+    gridir.add_aux_coord(lon_coord, data_dims=(0, 1))
+    fn = 'fx_test_areacello.nc'
+    iris.save(gridir, fn)
+    return fn
```

Comment:
@valeriupredoi I wrote some tests (`test_area_statistics_mean_ir` and test_area_statistics_sum_ir`) and they are failing because IRIS doesn't like the fx cube I'm making here:

Any thoughts?

Reply:
line 3277 from `cube.py` is trying to view (memory map) your **numpy** array but your weights is a dask array that doesn't have `view` as attribute:
```
In [42]: b                                                                                                                                        
Out[42]: dask.array<array, shape=(2,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>

In [43]: b.view()                                                                                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-43-4a91f95886bf> in <module>
----> 1 b.view()

TypeError: view() missing 1 required positional argument: 'dtype'
```
you need to realize your weights before calling the collapse (I don't think they've implemented lazy weights yet in iris)

Reply:
just replace:
```
@@ -322,7 +322,7 @@ def area_statistics(cube, operator, fx_files=None, calculate_grid=False):
                          '({})'.format(cube.shape, grid_areas.shape))
 
     return cube.collapsed(['longitude', 'latitude'], operation,
-                          weights=grid_areas)
+                          weights=np.array(grid_areas))
```
note that the two tests will fail again but this time the result is not the same as expected (first is 1.5 when you check on 1.0 and the second is 7.381511e+12 when you check on 1.0) - I suspect you are not adding the right results spat by area statistics

Reply:
I did something like this, but used:
```
    if isinstance(grid_areas, da.Array):
        grid_areas = grid_areas.compute()
```

Reply:
ya that'll do! bit confused as to why your area statistic is such a monster number (7e12)